File: tools/kill_process.lua
return {
    name = "kill_process",
    description = 'Kill a detached process. Input should be in JSON format: {"id": "1783687230"}',
    tool = {
        type = "function",
        ["function"] = {
            name = "kill_process",
            description = 'Kill a detached process. Input should be in JSON format: {"id": "1783687230"}',
            parameters = {
                type = "object",
                properties = {
                    id = {
                        type = "string",
                        description = "The process id to kill"
                    }
                },
                required = {"id"}
            }
        }
    },
    action = function(input)
        local id = input.id
        if not (AskProceed("kill_process")) then
            return { success = false, content = "Command execution denied by user" }
        end
        if _G.run_command[tostring(id)] then
        end
    end
}

File: tools/kill_process.py
"""kill_process tool — kill a detached process by ID (stub)."""

name = "kill_process"
description = 'Kill a detached process. Input: {"id": "pid"}'
schema = {
    "type": "function",
    "function": {
        "name": "kill_process",
        "description": 'Kill a detached process. Input: {"id": "pid"}',
        "parameters": {
            "type": "object",
            "properties": {
                "id": {"type": "string", "description": "The process ID to kill"},
            },
            "required": ["id"],
        },
    },
}


def action(input: dict) -> dict:
    from communication import ask_proceed
    if not ask_proceed("kill_process"):
        return {"success": False, "content": "Command execution denied by user"}
    # Stub — in the Lua version this was incomplete too
    return {"success": True, "content": "Process killed (stub)"}
